### ISSUE TITLE
Possible cascading issue B13?

### ISSUE THREAD

**Issue by SenpaiDerp** [Azanor/thaumcraft-beta#877]
I've been working on making a modpack for the past few months after your initial release to 1.12.  I am currently running B13 because B15 simply crashes every time I enter a magic forest biome due to a particle error.  My issue is that, when ever I encounter a LARGE magic forest, my client seems to lock up and spam my logs with cascading worldgen.  Does the mod use this feature?  I know sometimes forge misinterprets what is cascading and what isn't, but it does consistently happen in larger magic forests for me.  If it does use this feature, is there a way to limit how many chunks it will load outside of my render distance?

I have tested this a few times and re-logging normally fixes cascading issues, but not if you log out in or near the larger forests; small forests seem fine which is why I ask about possible cascades.  Let me know if you have any ideas :D

**Comment by Geethebluesky** [comment]
Was this a new magical forest you'd never visited before, or someplace you'd seen already?

Cascading worldgen usually happens when mods try to generate structures. If you can, try to note the coordinates of a few of the chunks that are listed in the error logs and see if anything's generated there.

**Comment by SenpaiDerp** [comment]
It happened when I first visited, as I stayed in it and finished loading the rest, and after I re-logged to see if it would clear excess render a bit.

I knew some parts of the mod used the method, but I wasn't sure if forests may aswell which would explain why such a large forest gave so much of an issue.

I am not 100% sure if this was the issue or not, just something I saw in the logs and did some excess research on and noticed that it really only happened in the huge forests.  If I am wrong though, I do apologize!

**Comment by Azanor** [comment]
Yeah, currently greatwoods do cause a bit of cascading worldgen, but nothing too gamebreaking. There is another bug related to auras however that is causing something similar and which is much nastier. It will be fixed in B16.
